core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceBillingAccountActor.scala

```diff
@@ -0,0 +1,275 @@
+        DBIO.seq(
+          thisChange.setGoogleSyncTime(syncTime.some),
+          thisChange.setOutcome(outcome.some),
+          thisBillingProject.setInvalidBillingAccount(!billingProbeCanAccessBillingAccount.getOrElse(true)),
+          thisBillingProject.setMessage(message)
+        )
+      }.io
+    } yield ()
+
+
+  def updateWorkspacesInProject(billingAccountChange: BillingAccountChange,
+                                billingProject: RawlsBillingProject,
+                                billingProjectSyncOutcome: Outcome): IO[Unit] =
+    for {
+      // v1 workspaces use the v1 billing project's google project and we've already attempted
+      // to update which billing account it uses. We can update, therefore, all workspaces that
+      // use that google project with the outcome of setting the billing account on the v1
+      // billing project.
+      _ <- setWorkspaceBillingAccountAndErrorMessage(
+        workspaceQuery
+          .withBillingProject(billingProject.projectName)
+          .withGoogleProjectId(billingProject.googleProjectId),
+        billingProject.billingAccount,
+        Outcome.toTuple(billingProjectSyncOutcome)._2
+      )
+
+      // v2 workspaces have their own google project so we'll need to attempt to set the
+      // billing account on each.
+      v2Workspaces <- listV2WorkspacesInProject(billingProject)
+      _ <- v2Workspaces.traverse_ { workspace =>
+        for {
+          _ <- info("Updating Billing Account on Workspace Google Project",
+            "changeId" -> billingAccountChange.id,
+            "workspace" -> workspace.toWorkspaceName,
+            "newBillingAccount" -> billingAccountChange.newBillingAccount.map(_.value)
+          )
+
+          workspaceSyncAttempt <- updateBillingAccountOnGoogle(
+            workspace.googleProjectId,
+            billingProject.billingAccount
+          ).attempt
+
+          _ <- recordV2WorkspaceSyncOutcome(
+            billingAccountChange,
+            workspace,
+            billingProject.billingAccount,
+            Outcome.fromEither(workspaceSyncAttempt)
+          )
+        } yield ()
+      }
+    } yield ()
+
+
+  private def listV2WorkspacesInProject(billingProject: RawlsBillingProject): IO[List[Workspace]] =
+    dataSource.inTransaction { _ =>
+      workspaceQuery
+        .withBillingProject(billingProject.projectName)
+        .withoutGoogleProjectId(billingProject.googleProjectId)
+        .read
+    }
+      .io
+      .map(_.toList)
+
+
+  private def updateBillingAccountOnGoogle(googleProjectId: GoogleProjectId, newBillingAccount: Option[RawlsBillingAccountName]): IO[Unit] =
+    for {
+      projectBillingInfo <- gcsDAO.getBillingInfoForGoogleProject(googleProjectId).io
+      currentBillingAccountOnGoogle = getBillingAccountOption(projectBillingInfo)
+      _ <- Applicative[IO].whenA(newBillingAccount != currentBillingAccountOnGoogle) {
```

Comment:
All the changes are recorded in the database and we're logging the record id. I think we logged everything because we didnt persist this information anywhere. Now we are writing to the DB i think we can get away with logging less. What do you think?

Reply:
I think we log the old billing account according to Rawls, but not the billing account according to Google, right?